tests/v2/validation/provisioning/rke2/ace_test.go

```diff
@@ -0,0 +1,215 @@
+						machinePoolConfig := provider.MachinePoolFunc(generatedPoolName, namespace)
+
+						machineConfigResp, err := client.Steve.SteveType(provider.MachineConfigPoolResourceSteveType).Create(machinePoolConfig)
+						require.NoError(r.T(), err)
+
+						machinePools := machinepools.RKEMachinePoolSetup(nodeRoles0, machineConfigResp)
+
+						cluster := clusters.NewK3SRKE2ClusterConfig(clusterName, namespace, cni, cloudCredential.ID, kubeVersion, machinePools)
+
+						cluster.Spec.LocalClusterAuthEndpoint = r.localClusterAuthEndpoint
+
+						clusterResp, err := clusters.CreateK3SRKE2Cluster(client, cluster)
+						require.NoError(r.T(), err)
+
+						adminClient, err := rancher.NewClient(client.RancherConfig.AdminToken, client.Session)
+						require.NoError(r.T(), err)
+						kubeProvisioningClient, err := adminClient.GetKubeAPIProvisioningClient()
+						require.NoError(r.T(), err)
+
+						result, err := kubeProvisioningClient.Clusters(namespace).Watch(context.TODO(), metav1.ListOptions{
+							FieldSelector:  "metadata.name=" + clusterName,
+							TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+						})
+						require.NoError(r.T(), err)
+
+						checkFunc := clusters.IsProvisioningClusterReady
+
+						err = wait.WatchWait(result, checkFunc)
+						assert.NoError(r.T(), err)
+						assert.Equal(r.T(), clusterName, clusterResp.ObjectMeta.Name)
+						assert.Equal(r.T(), kubeVersion, cluster.Spec.KubernetesVersion)
+
+						clusterIDName, err := clusters.GetClusterIDByName(adminClient, clusterName)
+						assert.NoError(r.T(), err)
+
+						err = nodestat.IsNodeReady(client, clusterIDName)
+						require.NoError(r.T(), err)
+
+						kubeConfig, err := kubeconfig.GetKubeconfig(client, clusterIDName)
+						require.NoError(r.T(), err)
+
+						original, err := r.client.SwitchContext(clusterResp.Name, kubeConfig)
+						require.NoError(r.T(), err)
+
+						originalResp, err := original.Resource(corev1.SchemeGroupVersion.WithResource("pods")).Namespace("").List(context.TODO(), metav1.ListOptions{})
+						require.NoError(r.T(), err)
+						for _, pod := range originalResp.Items {
+							r.T().Logf("Pod %v", pod.GetName())
+						}
+
+						contexts, err := kubeconfig.GetContexts(kubeConfig)
+						require.NoError(r.T(), err)
+						var contextNames []string
+						for context := range contexts {
+							if strings.Contains(context, "pool") {
+								contextNames = append(contextNames, context)
+							}
+						}
+
+						for _, contextName := range contextNames {
+							dynamic, err := r.client.SwitchContext(contextName, kubeConfig)
+							require.NoError(r.T(), err)
+							resp, err := dynamic.Resource(corev1.SchemeGroupVersion.WithResource("pods")).Namespace("").List(context.TODO(), metav1.ListOptions{})
+							require.NoError(r.T(), err)
+							r.T().Logf("Switched Context to %v", contextName)
+							for _, pod := range resp.Items {
+								r.T().Logf("Pod %v", pod.GetName())
+							}
+						}
```

Comment:
Just another small nit as a follow-up this: https://github.com/rancher/rancher/pull/40273#discussion_r1185298575, would it make sense to use `assert` instead `require` in this loop of the context changes (L197 & L199)? Just to make sure even if one of the context changes fails we're still able to see other changes ultimately.

Reply:
agreed

Reply:
Changed to `assert.NoError`